src/libvirt-xml-create.js

```diff
@@ -233,6 +236,39 @@ export function getSnapshotXML(name, description) {
         snapElem.appendChild(descriptionElem);
     }
 
+    if (isExternal) {
+        if (memoryPath) {
+            const memoryElem = doc.createElement('memory');
+            memoryElem.setAttribute('snapshot', 'external');
+            memoryElem.setAttribute('file', memoryPath);
+            snapElem.appendChild(memoryElem);
+        }
+
+        const disksElem = doc.createElement('disks');
+        disks.forEach(disk => {
+            // Disk can have attribute "snapshot" set to "no", which means no snapshot should be created of the said disk
+            // This cannot be configured through cockpit, but we should uphold it nevertheless
+            // see "snapshot" attribute of <disk> element at https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms
+            if (disk.snapshot !== "no") {
+                const diskElem = doc.createElement('disk');
+                diskElem.setAttribute('name', disk.target);
+                diskElem.setAttribute('snapshot', 'external');
+
+                if (disk.type === "volume") {
+                    const poolPath = getStoragePoolPath(storagePools, disk.source.pool, connectionName);
+                    if (poolPath) {
+                        const sourceElem = doc.createElement('source');
+                        sourceElem.setAttribute('file', `${poolPath}/${disk.source.volume}.snap`);
+                        diskElem.appendChild(sourceElem);
+                    }
+                }
+
+                disksElem.appendChild(diskElem);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20240125-124532-d8538f72-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)